descriptor/qsarmolecular/src/test/java/org/openscience/cdk/qsar/descriptors/molecular/JPlogPDescriptorTest.java

```diff
@@ -0,0 +1,115 @@
+		IAtomContainer struct = parseSmiles("CCC(=O)O");
+		JPlogPDescriptor desc = new JPlogPDescriptor();
+		DescriptorValue answer = desc.calculate(struct);
+		DoubleResult result = (DoubleResult) answer.getValue();
+		double output = result.doubleValue();
+		assertEquals(0.3, output, 0.1);
+	}
+
+	@Test
+	public void testAcetonitrile() throws CDKException {
+		parser = new SmilesParser(SilentChemObjectBuilder.getInstance());
+		IAtomContainer struct = parseSmiles("CC#N");
+		JPlogPDescriptor desc = new JPlogPDescriptor();
+		DescriptorValue answer = desc.calculate(struct);
+		DoubleResult result = (DoubleResult) answer.getValue();
+		double output = result.doubleValue();
+		assertEquals(0.4, output, 0.1);
+	}
+
+	@Test
+	public void testAniline() throws CDKException {
+		parser = new SmilesParser(SilentChemObjectBuilder.getInstance());
+		IAtomContainer struct = parseSmiles("Nc1ccccc1");
+		JPlogPDescriptor desc = new JPlogPDescriptor();
+		DescriptorValue answer = desc.calculate(struct);
+		DoubleResult result = (DoubleResult) answer.getValue();
+		double output = result.doubleValue();
+		assertEquals(1.2, output, 0.1);
+	}
+
+	@Test
+	public void testFluorobenzene() throws CDKException {
+		parser = new SmilesParser(SilentChemObjectBuilder.getInstance());
+		IAtomContainer struct = parseSmiles("Fc1ccccc1");
+		JPlogPDescriptor desc = new JPlogPDescriptor();
+		DescriptorValue answer = desc.calculate(struct);
+		DoubleResult result = (DoubleResult) answer.getValue();
+		double output = result.doubleValue();
+		assertEquals(2.0, output, 0.1);
+	}
+
+	@Test
+	public void testSimpleTextFields() {
+		JPlogPDescriptor desc = new JPlogPDescriptor();
+		DescriptorSpecification specification = desc.getSpecification();
+		String name = desc.getDescriptorNames()[0];
+		assertEquals("JPlogP developed at Lhasa Limited www.lhasalimited.org",
+				specification.getSpecificationReference());
+		assertEquals("Jeffrey Plante - Lhasa Limited", specification.getImplementationVendor());
+		assertEquals("JPLogP", name);
+		assertEquals(1, desc.getParameterNames().length);
+		assertEquals("addImplicitH", desc.getParameterNames()[0]);
+	}
+
+	@Test
+	public void testGetHologram() throws CDKException {
+		JPlogPDescriptor desc = new JPlogPDescriptor();
+		parser = new SmilesParser(SilentChemObjectBuilder.getInstance());
+		IAtomContainer molecule = parseSmiles("c1ccccc1");
+		Map<Integer, Integer> holo = desc.getMappedHologram(molecule);
+		assertEquals(2, holo.keySet().size());
+		assertEquals(6, holo.get(106204).intValue());
+	}
+
+
+
+	private static IAtomContainer parseSmiles(String smiles) throws CDKException {
+		parser = new SmilesParser(SilentChemObjectBuilder.getInstance());
+		IAtomContainer molecule = parser.parseSmiles(smiles);
+		AtomContainerManipulator.percieveAtomTypesAndConfigureUnsetProperties(molecule);
+		CDKHydrogenAdder hAdder = CDKHydrogenAdder.getInstance(molecule.getBuilder());
+		hAdder.addImplicitHydrogens(molecule);
```

Comment:
Mmm... this should not actually be needed... I think this is actually done by the `SmilesParser` already. @johnmay ?

Reply:
I tried removing all three lines, but the tests broke due to the aromaticity. I've instead removed line 2 and 3 in the above excerpt but left in line 1 to ensure correct aromaticity perception.